### PR TITLE
Apply EditorConfig rules and lint Sphinx docs

### PR DESCRIPTION
**Pull request by szepeviktor**
This is the part of [the Integrity workflow](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/reusable-integrity.yml).
(a collection of antidotes to all problems I've ever met)

Using EditorConfig rules for all files,
except reStructuredText that is checked with https://github.com/sphinx-contrib/sphinx-lint
`sphinx-lint --enable all --max-line-length 220 docs/`

[off] This guy (in the code) is the enemy of lower-ascii-only-viktor: `¯\_(ツ)_/¯`

# ~~`"`~~

ps. my benefit from this PR: meeting sphinx-lint 🎉 it goes into my list of [default tools](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/sphinx.yml)

### CODE REVIEW

File: docs/authentication.rst [hunk 1]
```diff
@@ -320,8 +320,8 @@ each time to update the session cookie on expiration:
 
 * Improved security via JWT-based session tokens that can only be generated using authorized service accounts.
 * Stateless session cookies that come with all the benefit of using JWTs for authentication. The session cookie has
-  the same claims (including custom claims) as the ID token, making the same permissions checks enforceable on the
-  session cookies.
+    the same claims (including custom claims) as the ID token, making the same permissions checks enforceable on the
+    session cookies.
```
Comment (szepeviktor) [inline docs/authentication.rst:324]: Oh no!
This makes this
image
into this
image

Comment (szepeviktor) [reply docs/authentication.rst:324]: Should rst and md files be indented with 2 spaces?

**Comment by jeromegamez** [comment]
Thank you (honestly!) for the time spending on this, but in my view, whitespaces in the docs (and one duplicated `//` in a test) are not an issue/violation that needs fixing.

The relevant portions of the code are handled by PHP CS Fixer, and that's where I see the block comment rules, not in the editorconfig. PHP CS Fixer isn't part in the CI pipeline on purpose as well, I run them locally periodically and haven't had a problem with the PRs received so far, which I often times merge manually instead of in the GitHub UI.

I prefer the lines in the docs (.md or .rst) to be aligned instead of following the .editorconfig by the book.

So, in summary, in this project, the .editorconig is a hint, not a lint. I do many things manually instead of automatically because I prefer it that way - if something slips through (by creating view issues), I'll fix it after the fact.

If you want to add the `indent_size=2` to the `.md` block and remove the duplicate `//` in the test, and run your tools over the files manually, I'll revisit and merge the PR, but in its current form it's too much for me 😅

(I have never heard of the integrity workflow or eclint before - the jednado repo has been archived three years ago, and the greut repo has 8 stars, is this an industry standard tool that I have missed? Honest question!)

**Comment by szepeviktor** [comment]
Thank you for you long response.

> but in its current form it's too much for me 😅

I care about bytes every day all day.
This is the basis of every repository. If there there is turmoil in the foundation you cannot build a proper house on it.

> is this an industry standard tool that I have missed?

The industry standard is that no one cares about bytes!!
People are going for the benefit, bytes are only touched when there is a _must_.
I like keeping all my bytes (0-255) clean and happy.

Resending this PR ...

**Comment by szepeviktor** [comment]
@jeromegamez Done 🍏

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Troubleshooting
 
 .. note::
     This SDK works with immutable objects until noted otherwise. You can recognize these
-    objects when they have a ˚`with*``method. In that case, please keep in mind that in
+    objects when they have a ``with*`` method. In that case, please keep in mind that in
```
Comment (szepeviktor) [inline docs/troubleshooting.rst:7]: Here were two stray characters from a French keyboard 🇫🇷

File: docs/user-management.rst [hunk 3]
```diff
@@ -222,10 +222,10 @@ Any combination of the following properties can be provided:
 ================= ======= ===========
 Property          Type    Description
 ================= ======= ===========
-``uid``	          string  The uid to assign to the newly created user. Must be a string between 1 and 128 characters long, inclusive. If not provided, a random uid will be automatically generated.
+``uid``           string  The uid to assign to the newly created user. Must be a string between 1 and 128 characters long, inclusive. If not provided, a random uid will be automatically generated.
```
Comment (szepeviktor) [inline docs/user-management.rst:225]: TAB characters in reStructuredText 🤯

File: .editorconfig [hunk 4]
```diff
@@ -8,12 +8,19 @@ end_of_line = lf
 charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
+block_comment_start = /**
+block_comment = *
+block_comment_end = */
```
Comment (jeromegamez) [inline .editorconfig]: Could you remove these, please?

File: docs/troubleshooting.rst (hunk 2 above)
Comment (jeromegamez) [reply docs/troubleshooting.rst:7]: Haha 😂 , good catch!

File: docs/user-management.rst (hunk 3 above)
Comment (jeromegamez) [reply docs/user-management.rst:225]: How did that end up here! 🙈

**Review by jeromegamez — APPROVED** [review]
Thank you for the changes. The maximum line length in the docs doesn't change the result, but I'll give you that after rejecting the other changes :D.

But joking aside, thank you, also for the interesting discussion! `sphinx-lint` is new to me as well, that's a nice one - perhaps I will enforce coding/docs standards after all. I refrained from it mainly to not put off possible contributors, but if I'm honest, they are rare anyway 😅

Just one change (except you can convince me otherwise) and I think this is ready to merge 💪🏻

File: .editorconfig (hunk 4 above)
Comment (szepeviktor) [reply .editorconfig]: Removing right away 🏃🏻‍♂️ 
(the checker tool `eclint` needs to know about block comment signs as block comments contain an extra leading space)

Comment (szepeviktor) [reply .editorconfig]: Removed.

**Comment by szepeviktor** [comment]
You can have my byte patting tools with daily updates.
https://github.com/szepeviktor/byte-level-care/tree/master/.github/workflows

**Comment by jeromegamez** [comment]
> You can have my byte petting tools with daily updates. [szepeviktor/byte-level-care@`master`/.github/workflows](https://github.com/szepeviktor/byte-level-care/tree/master/.github/workflows?rgh-link-date=2023-10-28T19%3A18%3A39Z)

You're fast!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Comment by jeromegamez** [comment]
Thank you!

**Pull request merged**